src/arrays/referencing_array_and_its_elements/hard/q3/MainTest.java

```diff
@@ -0,0 +1,103 @@
+package arrays.referencing_array_and_its_elements.hard.q3;
+
+import global.BaseTest;
+import global.MethodTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.CustomAssertions;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.PlaceHolder;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.utils.ArrayUtil.arrayToInput;
+import static global.utils.ArrayUtil.generateRandomArray;
+
+public class MainTest extends BaseTest {
+    // Java
+
+    public static int n = 10;
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = arrayToInput(generateRandomArray(-100, 100, n));
+        return new Clause[]{
+                new StringLiteral("Enter " + n + " positive integers:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new StringLiteral("The array with only even numbers is:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new PlaceHolder(),
+                new Optional(new StringLiteral(" ")),
+        };
+    }
+
+    public void runMain() {
+        Even.main(new String[0]);
+    }
+
+    public static int[] getEven(int[] arr) {
+        int numEven = 0;
+        for (int i = 0; i < arr.length; i++)
+            numEven += arr[i] % 2 == 0 ? 1 : 0;
+        int[] evenNumbers = new int[numEven];
+        int evenIndices = 0;
+        for (int i = 0; i < arr.length; i++)
+            if (arr[i] % 2 == 0)
+                evenNumbers[evenIndices++] = arr[i];
+        return evenNumbers;
+    }
+
+    static Stream<Arguments> mainInputProvider() {
+        int[] a1 = generateRandomArray(1, 1000000000, n);
```

Comment:
We can add a case here where we have only evens and only odd numbers using array util